### PR TITLE
Reinstate the TestAggregateIntentWithPRClaims

### LINKED ISSUE #1296: Reinstate TestAggregateIntentWithPRClaims
TestAggregateIntentWithPRClaims in ics/keeper/intent_test.go is commented out. We should bring this test back into play and probably combine with TestAggregateIntent to ensure we have full coverage of intent aggregation.

This fits in with #1268 which will affect the IntentAggregation where there is a validator on the deny list.

### PR DESCRIPTION
**Pull request by tropicaldog**
## 1. Summary
Fixes #1296 

## 2.Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

## Summary by CodeRabbit

- **Refactor**
	- Enhanced test scenarios for intent aggregation in the interchain staking module.
	- Updated test functions to accurately represent functionalities related to validators and intents.
	- Adjusted variable declarations for improved test coverage.
	- Concealed internal details for confidentiality.

### CODE REVIEW

File: x/interchainstaking/keeper/intent_test.go [hunk 1]
```diff
@@ -352,218 +357,284 @@ func (suite *KeeperTestSuite) TestAggregateIntent() {
...
+				return map[string]sdk.Coins{user2.String(): sdk.NewCoins(sdk.NewCoin(denom, sdk.NewInt(1000)))}
+			},
+			expected: func(ctx sdk.Context, app *app.Quicksilver, zone icstypes.Zone) icstypes.ValidatorIntents {
+				out := icstypes.ValidatorIntents{}
+				out = append(out, &icstypes.ValidatorIntent{ValoperAddress: app.InterchainstakingKeeper.GetValidators(ctx, zone.ChainId)[0].ValoperAddress, Weight: sdk.OneDec()})
+				// only remote assets are considered, thus user2 balance is ignored...
+				// out = append(out, &icstypes.ValidatorIntent{ValoperAddress: app.InterchainstakingKeeper.GetValidators(ctx, zone.ChainId)[1].ValoperAddress, Weight: sdk.OneDec().Quo(sdk.NewDec(2))})
+				return out.Sort()
+			},
+		},
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/intent_test.go:433]: You are right, 'local' balances are no longer counted. A local balance is treated as a claim, with source as `ctx.ChainId`

e.g 
```
{UserAddress: user2.String(), ChainId: zone.ChainId, Module: cmtypes.ClaimTypeLiquidToken, Amount: sdkmath.NewInt(1000), SourceChainId: ctx.ChainId}
```

**Comment by joe-bowman** [comment]
Todo:

1) We should remove the `balances` field from testCases, and mint tokens based on the sum of claims.

2) We also need additional test cases. Presently, every case is practically identical - one user, with 100% of the supply being claimed by them. We should add:
- One user, multiple claims, sum == 100% of supply
- Two users, multiple claims, sum == 100% of supply
- Multiple users, multiple claims, sum is < 100% (i.e. not all users have claimed)

If the claims add up to 100% of supply, the aggregate intent should be the weighted average.
If the claims add up to < 100% of supply. the aggregate intent should be the weighted average for the claimed amount, and the default weightings (equal) for the unclaimed supply.
What happens if claim > 100% (this plausibly can happen, if users lend tokens on umee, they will claim, as with the borrower!)?

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>f6b5673</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/ADWaajXM91tiTA2dHTSNbuz8vnVb'>https://vercel.com/quicksilver/quicksilver/ADWaajXM91tiTA2dHTSNbuz8vnVb</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8440662340)

**Comment by tropicaldog** [comment]
> What happens if claim > 100% (this plausibly can happen, if users lend tokens on umee, they will claim, as with the borrower!)?.

@joe-bowman In this case, should we add a test using existing logic to handle this scenario, or is a code refactoring necessary?

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>74609c2</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/FHNYX28ArGM5sRQKivaAfFp3JTmZ'>https://vercel.com/quicksilver/quicksilver/FHNYX28ArGM5sRQKivaAfFp3JTmZ</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8447978834)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>0c286d2</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/FCaF67pB1w9bhSWDPnuRHqCx4nHA'>https://vercel.com/quicksilver/quicksilver/FCaF67pB1w9bhSWDPnuRHqCx4nHA</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8448047679)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>50ecfd4</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/G5FPcQsWEvATgsH3RiJLTpxAcSdS'>https://vercel.com/quicksilver/quicksilver/G5FPcQsWEvATgsH3RiJLTpxAcSdS</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8451254520)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>6b30f11</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/FBuyDEwYRqoRtoWUNKztSirJRzrw'>https://vercel.com/quicksilver/quicksilver/FBuyDEwYRqoRtoWUNKztSirJRzrw</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8461990216)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>e3bfe64</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/3BkQFEdgisCgfwo9pEWRoXmJhe3y'>https://vercel.com/quicksilver/quicksilver/3BkQFEdgisCgfwo9pEWRoXmJhe3y</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8463893610)

File: x/interchainstaking/keeper/intent_test.go [hunk 2]
```diff
@@ -352,218 +357,385 @@ func (suite *KeeperTestSuite) TestAggregateIntent() {
...
+			intents: func(ctx sdk.Context, app *app.Quicksilver, zone icstypes.Zone) []icstypes.DelegatorIntent {
+				out := make([]icstypes.DelegatorIntent, 0)
+				out = append(out, icstypes.DelegatorIntent{Delegator: user1.String(), Intents: icstypes.ValidatorIntents{&icstypes.ValidatorIntent{ValoperAddress: app.InterchainstakingKeeper.GetValidators(ctx, zone.ChainId)[0].ValoperAddress, Weight: sdk.OneDec()}}})
+				return out
+			},
+			claims: func(ctx sdk.Context, app *app.Quicksilver, zone icstypes.Zone) map[string]cmtypes.Claim {
+				return map[string]cmtypes.Claim{
+					user1.String(): {UserAddress: user1.String(), ChainId: zone.ChainId, Module: cmtypes.ClaimTypeLiquidToken, Amount: sdkmath.NewInt(1000), SourceChainId: "osmosis-1"},
+					user1.String(): {UserAddress: user1.String(), ChainId: zone.ChainId, Module: cmtypes.ClaimTypeLiquidToken, Amount: sdkmath.NewInt(1000), SourceChainId: "osmosis-1"},
+				}
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/intent_test.go:398]: Rabit flagged that only one claim is set here because they both use the same map index.

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>d86693c</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/36LSe3NsG6y5JcPpacxYtyXVWKiF'>https://vercel.com/quicksilver/quicksilver/36LSe3NsG6y5JcPpacxYtyXVWKiF</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8478140524)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>1dfd35c</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/HQypCkq5N63rzGLHHFFfszV5CEmA'>https://vercel.com/quicksilver/quicksilver/HQypCkq5N63rzGLHHFFfszV5CEmA</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8478141793)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>ddb0413</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1320.app-dev.quicksilver.zone'>https://pr-1320.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/8Qm3MTGN53C2KEPNVwgG3xahoobv'>https://vercel.com/quicksilver/quicksilver/8Qm3MTGN53C2KEPNVwgG3xahoobv</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8603556023)

**Pull request merged**